src/server/graphql/models/authorization.js

```diff
@@ -42,6 +43,32 @@ export const requireSUOrSelf = (authToken, userId) => {
   throw errorObj({_error: 'Unauthorized. You cannot modify another user.'});
 };
 
+export const requireSUOrSelfOrLead = async (authToken, userId, teamId) => {
```

Comment:
When we introduce payment, we might turn tms into an array of objects with different roles, but for now, it's easy enough to just check the DB.